src/ProxyClient/Http/HttpAdapterInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace FOS\HttpCache\ProxyClient\Http;
+
+use FOS\HttpCache\Exception\ExceptionCollection;
+
+/**
+ * An adapter to work with Httplug.
+ *
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+interface HttpAdapterInterface
```

Comment:
I agree, it should not be Http related at all. For example in FXMLRPC, http based client is just a transfer object. It could be SSH as well, but that's xmlrpc specific.

So this should be a generic interface with a naming which explains what it does (Invalidator?). The actual implementation can be Http dependent. Even if it does not make sense to invalidate without http, from an abstract point of view, it doesn't matter.